### PR TITLE
Guides 2.0 - Routes

### PR DESCRIPTION
**Pull request by swilgosz**
### Overview

Routes explanation for guides 2.0

- [x] Overview
- [x] Basic Usage
- [x] Advanced Usage
- [x] Testing

Relates to: #97

### CODE REVIEW

**Comment by swilgosz** [comment]
[Hanami-Guides: Router](https://trello.com/c/hWSRlmit/137-hanami-guides-router)

File: content/v2.0/router/advanced-usage.md [hunk 1]
```diff
@@ -0,0 +1,624 @@
...
+title: "Router: Advanced Usage"
+order: 30
+---
+
+
+[Refer](https://github.com/hanami/router/blob/unstable/spec/integration/hanami/router/mount_spec.rb)
+
+## Mounting an application
+
+In Hanami, you can mount any rack application in desired route. If we want to do it, you should use the `mount` keyword.
```
Comment (solnic) [inline content/v2.0/router/advanced-usage.md:11]:
```suggestion
In Hanami, you can mount any rack application under desired route, in order to do it, you can use `mount` method:
```

File: content/v2.0/router/advanced-usage.md [hunk 2]
```diff
@@ -0,0 +1,624 @@
...
+  Then your application can be visited at <code>http://blog.lvh.me:2300</code>
+</p>
+-->
+
+### RESTful Resources
+
+Hanami has native [REST](http://en.wikipedia.org/wiki/Representational_state_transfer) support.
+
+At the routing level, there are two methods that can be used to declare them: `resources` and `resource`.
+The former is for plural resources, the latter for singular ones.
```
Comment (solnic) [inline content/v2.0/router/advanced-usage.md:62]: I feel like this sentence is not needed.

File: content/v2.0/router/advanced-usage.md [hunk 3]
```diff
@@ -0,0 +1,624 @@
...
+  <tr>
+    <td>DELETE</td>
+    <td>/books/:id</td>
+    <td>Books::Destroy</td>
+    <td>:destroy</td>
+    <td>:book</td>
+  </tr>
+</table>
+
+### Remove Routes
```
Comment (solnic) [inline content/v2.0/router/advanced-usage.md:140]:
```suggestion
### Excluding RESTful routes
```

File: content/v2.0/router/advanced-usage.md [hunk 4]
```diff
@@ -0,0 +1,624 @@
...
+Hanami.applicaton.routes do
+  resources :books, only: [:new, :create, :show]
+
+  # equivalent to
+
+  resources :books, except: [:index, :edit, :update, :destroy]
+end
+```
+
+### Add Routes
```
Comment (solnic) [inline content/v2.0/router/advanced-usage.md:157]:
```suggestion
### Adding Routes
```

File: content/v2.0/router/advanced-usage.md [hunk 5]
```diff
@@ -0,0 +1,624 @@
...
+
+Alongside with default routes we can specify extra routes for single (`member`) or multiple (`collection`) resources.
```
Comment (solnic) [inline content/v2.0/router/advanced-usage.md:159]:
```suggestion
Alongside the default routes we can specify extra routes for single (`member`) or multiple (`collection`) resources.
```

**Review by solnic — CHANGES_REQUESTED** [review]
I've left some suggestions. On top of this, please pay attention to "we" vs "you" because it's inconsistent right now (here and in other guides). My understanding is that we want to always refer to "you" (the reader). So, ie "If you want to do X" rather than "If we want to do X".

**Comment by swilgosz** [comment]
Closing, as routing is already deployed with another PR.

**Pull request closed without merging**